Stratis.Bitcoin/Features/Consensus/ConsensusManager.cs

```diff
@@ -57,5 +58,12 @@ public Target GetNetworkDifficulty()
             else
                 return null;
         }
+
+        /// <inheritdoc />
+        public async Task<UnspentOutputs> GetUnspentTransactionAsync(uint256 trxid)
+        {
+            var outputs =  await this.ConsensusLoop.UTXOSet?.FetchCoinsAsync(new[] { trxid });
+            return outputs == null || outputs.UnspentOutputs == null || outputs.UnspentOutputs.Length == 0 ? null : outputs.UnspentOutputs[0];
```

Comment:
Shortcuts are nice but less readable, but if you already go down that path then this would work no?
```
return outputs?.UnspentOutputs?.SingleOrDefault()
```